docs/csharp/tutorials/exploration/csharp-7.yml

```
@@ -0,0 +1,536 @@
+  durationInMinutes: 2
+  content: |
+    The existing syntax that supports `out` parameters has been improved
+    in this version.  
+
+    Previously, you would need to separate the declaration of the out variable
+    and its initialization into two different statements:
+
+    [!code-csharp[OutVariableOldStyle](~/samples/snippets/csharp/new-in-7/program.cs#03_OutVariableOldStyle "classic out variable declaration")]
+
+    You can now declare `out` variables in the argument list of a method call,
+    rather than writing a separate declaration statement:
+
+    [!code-csharp[OutVariableDeclarations](~/samples/snippets/csharp/new-in-7/program.cs#01_OutVariableDeclarations "Out variable declarations")]
+
+    You may want to specify the type of the `out` variable for clarity,
+    as shown above. However, the language does support using an implicitly
+    typed local variable:
+
+    [!code-csharp[OutVarVariableDeclarations](~/samples/snippets/csharp/new-in-7/program.cs#02_OutVarVariableDeclarations "Implicitly typed Out variable")]
+
+    * The code is easier to read. 
+        - You declare the out variable where you use it, not on another line above.
+    * No need to assign an initial value.
+        - By declaring the `out` variable where it is used in a method call, you can't accidentally use it before it is assigned.
+
+    The most common use for this feature will be the `Try` pattern. In this
+    pattern, a method returns a `bool` indicating success or failure and an
+    `out` variable that provides the result if the method succeeds.
+
+    When using the `out` variable declaration, the declared variable "leaks" into the outer scope of the if statement. This allows you to use the variable afterwards:
+
+    ```csharp
+    if (!int.TryParse(input, out int result))
+    {    
+        return null;
+    }
+
+    return result;
+    ```
+
+- title: Tuples create light-weight data structures
+  durationInMinutes: 1
+  content: |
+    > [!NOTE]
+    > The new tuples features require the <xref:System.ValueTuple> types.
+    > You must add the NuGet package [`System.ValueTuple`](https://www.nuget.org/packages/System.ValueTuple/) in order to use it
+    > on platforms that do not include the types.
+    >
+    > This is similar to other language features that rely on types
+    > delivered in the framework. Example include `async` and `await`
+    > relying on the `INotifyCompletion` interface, and LINQ relying
+    > on `IEnumerable<T>`. However, the delivery mechanism is changing
+    > as .NET is becoming more platform independent. The .NET Framework
+    > may not always ship on the same cadence as the language compiler. When new language
+    > features rely on new types, those types will be available as NuGet packages when
+    > the language features ship. As these new types get added to the .NET Standard
+    > API and delivered as part of the framework, the NuGet package requirement will
```

```suggestion
    > Library and delivered as part of the framework, the NuGet package requirement will
```
framework, or .NET Framework?